staging/src/k8s.io/kubectl/pkg/cmd/apply/applyset_pruner.go

```diff
@@ -16,15 +16,114 @@ limitations under the License.
 
 package apply
 
-import "fmt"
+import (
+	"context"
+	"fmt"
+	"io"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/dynamic"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+)
 
 type applySetPruner struct {
+	dynamicClient dynamic.Interface
+
+	visitedUids sets.String
```

Comment:
When I pull this branch my IDE is letting me know that this is deprecated in favour of `sets.Set[string]`.

Reply:
Yes; to fix we'd also have to / really want to update pkg/cmd/apply though, because we copy the value from there.  Should I send a (separate?) PR to do that?  Do it here?